src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -103,24 +105,36 @@ internal sealed class RSession : IRSession, IRCallbacks {
 
         public Task<IRSessionInteraction> BeginInteractionAsync(bool isVisible = true, CancellationToken cancellationToken = default(CancellationToken)) {
             if (!_isHostRunning) {
-                return CanceledInteractionTask;
+                return CanceledBeginInteractionTask;
             }
 
             RSessionRequestSource requestSource = new RSessionRequestSource(isVisible, _contexts, cancellationToken);
```

Comment:
Good catch. Weird, though, that we haven't seem to run into this before. @AlexanderSher, is this a problem, or are we missing something?